RUFAS/input_manager.py

```diff
@@ -53,144 +77,368 @@ def _load_metadata(self, metadata_path: str = "input/example_metadata.json") ->
-                                   info_map)
-            except Exception as e:
-                raise e
+        om.add_log("open_csv_file", f"Attempting to open {file_path}.", info_map)
+        try:
+            with open(file_path, "r") as csv_file:
+                data_frame = pd.read_csv(csv_file)
+                data_dict = {column: data_frame[column].tolist() for column in data_frame.columns}
+                if not data_frame.empty:
+                    om.add_log("load_data_successful",
+                               f"Successfully loaded data from {file_path}.",
+                               info_map)
+                return data_dict
+        except Exception as e:
+            raise e
 
-    def _validate_data(self, eager_termination: bool = True) -> bool:
+    def _validate_data_and_add_to_pool(self, eager_termination: bool) -> bool:
         """
-        Validates input data and attempts to fix any invalid input data.
+        Validates data from input files and adds data to pool if valid.
 
         Parameters
         ----------
-        eager_termination : bool, default=True
-            If true, the process will be terminated upon finding invalid data.
+        eager_termination : bool
+            If True, the process will be terminated as soon as finding invalid data and failing to fix it.
+            If False, the process will be terminated after going through and validating the entire data,
+            if invalid data is found.
 
         Returns
         -------
         bool
-            True if all data is valid; False otherwise.
+            True if data is valid, otherwise False.
         """
         info_map = {"class": self.__class__.__name__,
-                    "function": self._validate_data.__name__,
+                    "function": self._validate_data_and_add_to_pool.__name__,
                     }
         valid_elements_counter = 0
         invalid_elements_counter = 0
+        total_elements_counter = 0
         fixed_elements_counter = 0
-        invalid_critical_elements_counter = 0
-        total_elements_checked_counter = 0
-
-        for key in self.__pool.keys():
-            for variable, value in self.__pool[key].items():
-                total_elements_checked_counter += 1
-                if self._validate_element(variable, value):
-                    valid_elements_counter += 1
+
+        for file_blob_key, file_details in self.__metadata["files"].items():
+            file_path = file_details["path"]
+            data_type_to_loader_map = {"json": self._load_data_from_json,
+                                       "csv": self._load_data_from_csv}
+            try:
+                data_loader = data_type_to_loader_map[file_details["type"]]
+                data = data_loader(file_path)
+            except KeyError:
+                raise KeyError(f"Faulty data type in {file_blob_key}, only CSV and JSON are supported.")
```

Comment:
I know I wrote this error message in an earlier comment, but now with the additional context I think this is better. So later on when we want to add new types we just need to implement its function and add its handle to the dict.
```suggestion
                raise KeyError(f"Faulty data type in {file_blob_key}, supported types are: {data_type_to_loader_map.keys()}.")
```

Reply:
I like it.